src/main/java/org/prebid/server/util/algorithms/random/RandomPositiveWeightedEntrySupplier.java

```diff
@@ -0,0 +1,20 @@
+package org.prebid.server.util.algorithms.random;
+
+import java.util.function.Function;
+
+public class RandomPositiveWeightedEntrySupplier<E> extends RandomAnyWeightedEntrySupplier<E> {
+
+    public RandomPositiveWeightedEntrySupplier(Function<E, Integer> weightExtractor) {
+        super(weightExtractor);
+    }
+
+    protected int weight(E entry) {
+        final int weight = weightExtractor.apply(entry);
+        if (weight > 0) {
+            return weight;
+        }
+
+        throw new IllegalArgumentException("Non-positive weight.");
```

Comment:
Better 'Model weight should be greater then 0'

Reply:
What is the "model" in this context? XD

Reply:
than `Wweight should be greater then 0` )